frontend/src/metabase/querying/components/FilterModal/utils/filters.ts

```diff
@@ -6,23 +6,10 @@ import {
 } from "metabase/common/utils/column-groups";
 import type { GroupItem } from "metabase/querying/components/FilterContent";
 import * as Lib from "metabase-lib";
-
-export function appendStageIfAggregated(query: Lib.Query) {
-  const aggregations = Lib.aggregations(query, -1);
-  const breakouts = Lib.breakouts(query, -1);
-
-  return aggregations.length > 0 && breakouts.length > 0
-    ? Lib.appendStage(query)
-    : query;
-}
-
-function getStageIndexes(query: Lib.Query) {
-  const stageCount = Lib.stageCount(query);
-  return stageCount > 1 ? [-2, -1] : [-1];
-}
+import { getFilterStageIndexes } from "metabase-lib/v1/parameters/utils/targets";
```

Comment:
There should be no such dependency.

Reply:
b14ea1bec7f392bd6c0dc2a4dc7a7ae630be741a